src/libs/actions/PaymentMethods.ts

```diff
@@ -228,40 +236,53 @@ function addSubscriptionPaymentCard(cardData: {
     const optimisticData: OnyxUpdate[] = [
         {
             onyxMethod: Onyx.METHOD.MERGE,
-            key: ONYXKEYS.FORMS.ADD_DEBIT_CARD_FORM,
+            key: ONYXKEYS.FORMS.ADD_PAYMENT_CARD_FORM,
             value: {isLoading: true},
         },
     ];
 
     const successData: OnyxUpdate[] = [
         {
             onyxMethod: Onyx.METHOD.MERGE,
-            key: ONYXKEYS.FORMS.ADD_DEBIT_CARD_FORM,
+            key: ONYXKEYS.FORMS.ADD_PAYMENT_CARD_FORM,
             value: {isLoading: false},
         },
     ];
 
     const failureData: OnyxUpdate[] = [
         {
             onyxMethod: Onyx.METHOD.MERGE,
-            key: ONYXKEYS.FORMS.ADD_DEBIT_CARD_FORM,
+            key: ONYXKEYS.FORMS.ADD_PAYMENT_CARD_FORM,
             value: {isLoading: false},
         },
     ];
 
-    // TODO integrate API for subscription card as a follow up
-    API.write(WRITE_COMMANDS.ADD_PAYMENT_CARD, parameters, {
-        optimisticData,
-        successData,
-        failureData,
-    });
+    if (currency === CONST.CURRENCY.GBP) {
+        // eslint-disable-next-line rulesdir/no-api-side-effects-method
+        API.makeRequestWithSideEffects(SIDE_EFFECT_REQUEST_COMMANDS.ADD_PAYMENT_CARD_GBR, parameters, {optimisticData, successData, failureData}).then((response) => {
```

Comment:
Can you point me to where we decided to use `makeRequestWithSideEffects`? Is it in the design doc? Or a slack thread? Sorry if I lack context, still learning about this project after taking the lead on it a couple weeks ago. We do really want to avoid using this when at all possible, so just want to know where we made this decision so I can better understand why this was written this way.

Reply:
@blimpich yeah - this is way where we can grab particular item from response. We just need to grap authenticationLink to start iframe process
Here in doc:
`After pressing the “Add a payment card” button we will check the currency and for GBP call AddPaymentCardGBP API with data (cardNumber, cardMonth, cardYear, cardCVV, addressName, addressZip, currency). In response, we will receive an authenticationLink that we will render in the <iframe />. (we will utilize the approach from WalletStatementModal)`
But mostly logic will be covered in other follow up issue here https://github.com/Expensify/App/pull/44321/files

Reply:
I see, makes sense, thank you for the explanation!